docs/standard/native-interop/tutorial-comwrappers.md

```diff
@@ -0,0 +1,430 @@
+
+```csharp
+IntPtr iunk = ...; // Get a COM instance from native code.
+object rcw = cw.GetOrCreateObjectForComInstance(iunk, CreateObjectFlags.UniqueInstance);
+```
+
+### Step 4 &ndash; Handle Native Object Wrapper lifetime details
+
+The `ComputeVtables()` and `CreateObject()` implementations covered some wrapper lifetime details, but there are further considerations. While this can be a short step, it can also significantly increase the complexity of the `ComWrappers` design.
+
+Unlike the Managed Object Wrapper, which is controlled by calls to its `AddRef()` and `Release()` methods, the lifetime of a Native Object Wrapper is nondeterministically handled by the GC. The question here is, when does the Native Object Wrapper call `Release()` on the `IntPtr` that represents the COM instance? There are two general buckets:
+
+1) The Native Object Wrapper's Finalizer is responsible for calling the COM instance's `Release()` method. This is the only time when it's safe to call this method. At this point, it's been correctly determined by the GC that there are no other references to the Native Object Wrapper in the .NET runtime. There can be complexity here if you're properly supporting COM Apartments; for more information, see the [Additional considerations](#additional-considerations) section.
+
+2) The Native Object Wrapper implements `IDisposable` and calls `Release()` in `Dispose()`.
+
+  > [!NOTE]
+  > The `IDisposable` pattern should only be supported if, during the `CreateObject()` call, the `CreateObjectFlags.UniqueInstance` flag was passed in. If this requirement is not followed, it's possible for disposed Native Object Wrappers to be reused after being disposed.
+
+## Using the `ComWrappers` subclass
+
+You now have a `ComWrappers` subclass that can be tested. To avoid creating a native library that returns a COM instance that implements `IDemoGetType` and `IDemoStoreType`, you'll use the Managed Object Wrapper and treat it as a COM instance &ndash; this must be possible in order to pass it COM anyways.
+
+Let's create a Managed Object Wrapper first. Instantiate a `DemoImpl` instance and display its current string state.
+
+```csharp
+var demo = new DemoImpl();
+
+string? value = demo.GetString();
+Console.WriteLine($"Initial string: {value ?? "<null>"}");
+```
+
+Now you can create an instance of `DemoComWrappers` and a Managed Object Wrapper that you can then pass into a COM environment.
+
+```csharp
+var cw = new DemoComWrappers();
+
+IntPtr ccw = cw.GetOrCreateComInterfaceForObject(demo, CreateComInterfaceFlags.None);
+```
+
+Instead of passing the Managed Object Wrapper to a COM environment, pretend you just received this COM instance, so you'll create a Native Object Wrapper for it instead.
+
+```csharp
+var rcw = cw.GetOrCreateObjectForComInstance(ccw, CreateObjectFlags.UniqueInstance);
```

Comment:
I think both `GetOrCreateComInterfaceForObject` and `GetOrCreateObjectForComInstance` should document their behavior with regard to `AddRef`/`Release` on the native pointer. For example in this case, I honestly can't tell if the code should release `ccw` after it's used here, or if the reference is "Taken over" by the RCW creation. (COM conventions seem to dictate that the `ccw` should be released in this case, so maybe this is a leak?)